### DISCUSSION TITLE
CPU utlization drops in case of keda scale down.

### DISCUSSION THREAD

**Discussion by jeganathanp** [kedacore/keda#4593] [Q&A / Need Help] [answered]
We are currently operating a CPU and IO intensive application that experiences fluctuating load patterns. Approximately every 15 minutes, we encounter peak loads followed by a period of reduced activity lasting around 10 minutes.

To manage resource scaling efficiently, we have implemented Keda along with an autoscaler. However, we have encountered a specific issue when scaling in, where some "Pending" pods occupy space, leading to a decrease in CPU utilization. Despite our efforts to optimize this situation, we have not achieved significant success.

At present, we are utilizing an external script to handle the cleanup of pending pods, but its impact has been limited. We are exploring the possibility of an inherent solution for this problem.

Any suggestion is welcome.

**Comment by zroubalik** [comment, marked as answer]
I think we can't do anything about it from KEDA. Please check that your deployment handles termination correctly.
